storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorData.java

```diff
@@ -0,0 +1,234 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.daemon.supervisor;
+
+import org.apache.storm.Config;
+import org.apache.storm.StormTimer;
+import org.apache.storm.cluster.ClusterStateContext;
+import org.apache.storm.cluster.ClusterUtils;
+import org.apache.storm.cluster.DaemonType;
+import org.apache.storm.cluster.IStormClusterState;
+import org.apache.storm.daemon.supervisor.workermanager.IWorkerManager;
+import org.apache.storm.generated.LocalAssignment;
+import org.apache.storm.generated.ProfileRequest;
+import org.apache.storm.localizer.Localizer;
+import org.apache.storm.messaging.IContext;
+import org.apache.storm.scheduler.ISupervisor;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.LocalState;
+import org.apache.storm.utils.Utils;
+import org.apache.storm.utils.VersionInfo;
+import org.apache.zookeeper.data.ACL;
+import org.eclipse.jetty.util.ConcurrentHashSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class SupervisorData {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SupervisorData.class);
+
+    private final Map conf;
+    private final IContext sharedContext;
+    private volatile boolean active;
+    private ISupervisor iSupervisor;
+    private Utils.UptimeComputer upTime;
+    private String stormVersion;
+    private ConcurrentHashMap<String, String> workerThreadPids; // for local mode
+    private IStormClusterState stormClusterState;
+    private LocalState localState;
+    private String supervisorId;
+    private String assignmentId;
+    private String hostName;
+    // used for reporting used ports when heartbeating
+    private AtomicReference<Map<Long, LocalAssignment>> currAssignment;
+    private StormTimer heartbeatTimer;
+    private StormTimer eventTimer;
+    private StormTimer blobUpdateTimer;
+    private Localizer localizer;
+    private AtomicReference<Map<String, Map<String, Object>>> assignmentVersions;
+    private AtomicInteger syncRetry;
+    private final Object downloadLock = new Object();
+    private AtomicReference<Map<String, List<ProfileRequest>>> stormIdToProfileActions;
+    private ConcurrentHashSet<String> deadWorkers;
+    private final IWorkerManager workerManager;
```

Comment:
I think almost all of these member variables should be marked as final. The only ones I can see that are modified in a way final would not allow is active.